src/librbd/ImageCtx.h

```diff
@@ -230,6 +231,8 @@ namespace librbd {
     const SnapInfo* get_snap_info(librados::snap_t in_snap_id) const;
     int get_snap_name(librados::snap_t in_snap_id,
 		      std::string *out_snap_name) const;
+    int get_snap_namespace(librados::snap_t in_snap_id,
+			   cls::rbd::SnapshotNamespaceEntry *out_snap_namespace) const;
```

Comment:
Minor: could directly use ```SnapshotNamespace``` variant since ```SnapshotNamespaceEntry``` is only used for encode/decode and requires an indirection.